File: uikit/src/main/java/com/sendbird/uikit/widgets/TimelineMessageView.java
package com.sendbird.uikit.widgets;

import android.content.Context;
import android.content.res.TypedArray;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.sendbird.android.BaseMessage;
import com.sendbird.uikit.R;
import com.sendbird.uikit.databinding.SbViewTimeLineMessageComponentBinding;
import com.sendbird.uikit.utils.DateUtils;

public class TimelineMessageView extends BaseMessageView {
    private final SbViewTimeLineMessageComponentBinding binding;

    @NonNull
    public SbViewTimeLineMessageComponentBinding getBinding() {
        return binding;
    }

    @NonNull
    @Override
    public View getLayout() {
        return binding.getRoot();
    }

    public TimelineMessageView(@NonNull Context context) {
        this(context, null);
    }

    public TimelineMessageView(@NonNull Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, R.attr.sb_widget_timeline_message);
    }

    public TimelineMessageView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        TypedArray a = context.getTheme().obtainStyledAttributes(attrs, R.styleable.MessageView, defStyle, 0);
        try {
            this.binding = SbViewTimeLineMessageComponentBinding.inflate(LayoutInflater.from(getContext()), this, true);
            int textAppearance = a.getResourceId(R.styleable.MessageView_sb_message_timeline_text_appearance, R.style.SendbirdCaption1OnDark01);
            int backgroundResourceId = a.getResourceId(R.styleable.MessageView_sb_message_timeline_background, R.drawable.sb_shape_timeline_background);

            binding.tvTimeline.setTextAppearance(context, textAppearance);
            binding.tvTimeline.setBackgroundResource(backgroundResourceId);
        } finally {
            a.recycle();
        }
    }

    public void drawTimeline(@NonNull BaseMessage message) {
        binding.tvTimeline.setText(DateUtils.formatDate(message.getCreatedAt()));
    }
}

File: uikit/src/main/java/com/sendbird/uikit/internal/ui/messages/TimelineMessageView.kt
package com.sendbird.uikit.internal.ui.messages

import android.content.Context
import android.util.AttributeSet
import android.util.TypedValue
import android.view.LayoutInflater
import android.view.View
import com.sendbird.android.message.BaseMessage
import com.sendbird.uikit.R
import com.sendbird.uikit.databinding.SbViewTimeLineMessageComponentBinding
import com.sendbird.uikit.internal.extensions.setAppearance
import com.sendbird.uikit.internal.extensions.setTypeface
import com.sendbird.uikit.internal.model.notifications.NotificationConfig
import com.sendbird.uikit.utils.DateUtils
import com.sendbird.uikit.utils.DrawableUtils

internal class TimelineMessageView @JvmOverloads internal constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyle: Int = R.attr.sb_widget_timeline_message
) : BaseMessageView(context, attrs, defStyle) {
    override val binding: SbViewTimeLineMessageComponentBinding
    override val layout: View
        get() = binding.root

    init {
        val a = context.theme.obtainStyledAttributes(attrs, R.styleable.MessageView, defStyle, 0)
        try {
            binding = SbViewTimeLineMessageComponentBinding.inflate(
                LayoutInflater.from(getContext()),
                this,
                true
            )
            val textAppearance = a.getResourceId(
                R.styleable.MessageView_sb_message_timeline_text_appearance,
                R.style.SendbirdCaption1OnDark01
            )
            val backgroundResourceId = a.getResourceId(
                R.styleable.MessageView_sb_message_timeline_background,
                R.drawable.sb_shape_timeline_background
            )
            binding.tvTimeline.setAppearance(context, textAppearance)
            binding.tvTimeline.setBackgroundResource(backgroundResourceId)
        } finally {
            a.recycle()
        }
    }

    fun drawTimeline(message: BaseMessage) {
        binding.tvTimeline.text = DateUtils.formatTimelineMessage(message.createdAt)
    }

    fun drawTimeline(message: BaseMessage, uiConfig: NotificationConfig?) {
        binding.tvTimeline.run {
            text = DateUtils.formatTimelineMessage(message.createdAt)
            uiConfig?.let {
                val themeMode = it.themeMode
                it.theme.listTheme.timeline.apply {
                    setTextSize(TypedValue.COMPLEX_UNIT_SP, textSize.toFloat())
                    setTypeface(fontWeight.value)
                    measure(MeasureSpec.UNSPECIFIED, MeasureSpec.UNSPECIFIED)

                    DrawableUtils.createRoundedShapeDrawable(
                        backgroundColor.getColor(themeMode),
                        (measuredHeight / 2).toFloat()
                    ).apply { background = this }
                    setTextColor(textColor.getColor(themeMode))
                }
            }
        }
    }
}
